Weekly_Work/Week2/Code_Changes_Git.py

```diff
@@ -0,0 +1,159 @@
+import json
+import datetime
+from Commit import Commit
+import pandas as pd
+import matplotlib.pyplot as plt
+import utils
+
+class Code_Changes_Git(Commit):
+    """Class for Code_Changes for Git repositories.
+    
+    Objects are instantiated by specifying a file with the
+    commits obtained by Perceval from a set of repositories.
+        
+    :param path: Path to file with one Perceval JSON document per line
+    """
+    
+    def __init__(self, path, date_range=(None, None), source_code_exclude_list=None):
+        """
+        Initilizes self.df, the dataframe with one commit per row.
+        The source_code_exclude_list parameter is a list which contains file extensions
+        which are to be ignored. All possible file extensions are allowed. 
+        For files without a standard ".xyz" extension, like LICENCE or AUTHORS, the "others" 
+        extension is used. 
+        
+        :param path: Path to file with one Perceval JSON document per line
+        :param date_range: A tuple which represents the start and end date of interest
+        :param source_code_exclude_list: Files extensions to exclude. For example, 
+            source_code_exclude_list = ["py", "other", "gitignore"]
+        """
+        
+        super().__init__(path, date_range, source_code_exclude_list)
```

Comment:
I thought we had agreed on not having the code exclusion list...

Reply:
@jgbarah, in this patch, `source_code_exclude_list` is not limited to the excluded file extensions. A user can add paths like `tests/` or `xyz/abc/` to the list. As you had suggested, a class (`SourceCode`) was created, with different implementations of what source code means. 
Hence, there is a naive implementation, one based on file extensions and the last one based on directory paths. 
I thought this was what you were expecting. Alternatively, we could make the list an instance  member of the SourceCode class and omit the list from elsewhere. When running the code, the user will have to create a SourceCode object and pass in what things they want to exclude. 
This part of the project does require a bit more planning. Let me know what you think @aswanipranjal @valeriocos.